--- a/src/netxs/desktopio/lixx.hpp
+++ b/src/netxs/desktopio/lixx.hpp
@@ -34,6 +34,10 @@
 #include <sys/stat.h>    // ::fstat()
 #include <linux/input.h> // EV_*
 #include <fnmatch.h>     // ::fnmatch()
+#include <libgen.h>      // ::basename()
+#ifdef basename
+#undef basename
+#endif
 #include <dirent.h>      // ::dirent
 #include <fcntl.h>       // O_RDWR | O_NONBLOCK | O_CLOEXEC
 #include <sys/inotify.h> // ::inotify
