--- a/spacefm-master/src/common/meson.build
+++ b/spacefm-master/src/common/meson.build
@@ -6,7 +6,7 @@
     'logger',
     files('logger.cxx'),
     include_directories : include_directories('.'),
-    dependencies : spdlog_dep,
+    dependencies : [spdlog_dep, base_dependencies],
     install: false,
 )
 
@@ -63,6 +63,7 @@
 
 vfs_gtk3_dependencies = [
     gtk3_dependencies,
+    base_dependencies,
 
     logger_dep,
     libudev_dep,
@@ -130,6 +131,7 @@
 
 vfs_gtk4_dependencies = [
     gtk4_dependencies,
+    base_dependencies,
 
     logger_dep,
     libudev_dep,
@@ -162,6 +164,7 @@
     'natsort',
     natsort_sources,
     include_directories : include_directories('.'),
+    dependencies: base_dependencies,
     install: false,
 )
 
