--- a/spacefm-master/meson.build
+++ b/spacefm-master/meson.build
@@ -262,6 +262,7 @@
     sigc_dep,
 
     glycin_dep,
+    base_dependencies,
 ]
 
 ## preprocessor static
@@ -310,8 +311,10 @@
     glaze_dep = dependency('glaze', required: true, version: '>=6.3.0')
     base_dependencies += glaze_dep
 else
-    # TODO meson deprecated doing this
-    # incdir += include_directories('subprojects/glaze/include')
+    glaze_dep = declare_dependency(
+    include_directories: include_directories('subprojects/glaze/include'),
+)
+base_dependencies += glaze_dep
 endif
 
 if get_option('with-system-ztd')
@@ -334,6 +337,7 @@
     features += '-DSOCKET_PORT="@0@"'.format(get_option('zmp_port'))
 endif
 
+gtk4_dependencies += [ztd_dep, glaze_dep]
 media_dependencies = []
 if get_option('media')
     # ffmpeg
