--- a/spacefm-master/src/common/vfs/utils/editor.cxx
+++ b/spacefm-master/src/common/vfs/utils/editor.cxx
@@ -56,7 +56,7 @@
         return;
     }
 
-    const auto opened = desktop->open_files(path.parent_path(), {vfs::file::create(path)});
+    const auto opened = desktop->open_files(path.parent_path(), std::vector{vfs::file::create(path)});
     if (!opened)
     {
         gui::dialog::error(
